pennylane/resource/resource.py

```diff
@@ -63,6 +66,74 @@ class Resources:
     depth: int = 0
     shots: Shots = field(default_factory=Shots)
 
+    def __add__(self, other: Resources):
+        r"""Adds two :class:`~resource.Resources` objects together as if the circuits were executed in series.
+
+        Args:
+            other (Resources): the resource object to add
+
+        Returns:
+            Resources: the combined resources
+
+        .. details::
+
+            **Example**
+
+            >>> gate_types = {"Hadamard": 1, "CNOT": 1}
+            >>> gate_sizes = {1: 1, 2: 1}
+            >>> num_wires = 2
+            >>> depth = 2
+            >>> shots = qml.measurements.Shots(10)
+            >>> r1 = Resources(2, 2, gate_types, gate_sizes, 2, shots)
+            >>> gate_types = {"RX": 1, "CNOT": 1}
+            >>> gate_sizes = {1: 1, 2: 1}
+            >>> shots = qml.measurements.Shots((5, (2, 10))
+            >>> r2 = Resources(3, 2, gate_types, gate_sizes, 1, shots)
+            >>> print(r1 + r2)
+            wires: 3
+            gates: 4
+            depth: 3
+            shots: Shots(total=35, vector=[10 shots, 5 shots, 2 shots x 10])
+            gate_types:
+            {'Hadamard': 1, 'CNOT': 2, 'RX': 1}
+            gate_sizes:
+            {1: 2, 2: 2}
+        """
+        return add_in_series(self, other)
+
+    def __mul__(self, scalar: int):
```

Comment:
Let's add examples using the dunder methods somewhere. How about the [Resources documentation](https://docs.pennylane.ai/en/stable/code/api/pennylane.resource.Resources.html)?